test/adaptive_load/fake_plugins/fake_metrics_plugin/fake_metrics_plugin_test.cc

```diff
@@ -0,0 +1,146 @@
+#include "api/adaptive_load/benchmark_result.pb.h"
+#include "api/client/options.pb.h"
+#include "envoy/registry/registry.h"
+#include "adaptive_load/plugin_loader.h"
+#include "test/adaptive_load/fake_plugins/fake_metrics_plugin/fake_metrics_plugin.h"
+
+#include "external/envoy/source/common/config/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+namespace {
+
+using ::nighthawk::adaptive_load::BenchmarkResult;
+using ::nighthawk::adaptive_load::FakeMetricsPluginConfig;
+using ::nighthawk::adaptive_load::MetricEvaluation;
+using ::nighthawk::client::CommandLineOptions;
+using ::testing::HasSubstr;
+
+TEST(FakeMetricsPluginConfigFactory, CreateEmptyConfigProtoCreatesCorrectType) {
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<MetricsPluginConfigFactory>(
+          "nighthawk.fake-metrics-plugin");
+  Envoy::ProtobufTypes::MessagePtr message = config_factory.createEmptyConfigProto();
+  FakeMetricsPluginConfig expected_config;
+  EXPECT_EQ(message->DebugString(), expected_config.DebugString());
```

Comment:
We may want to double down with a MessageUtil bases comparison. Maybe we should consider a shared test helper utility for this.

Reply:
Sorry, I forgot all about MessageUtil.

I don't know how to create magic test macros but filed #467 to look into that.

Reply:
Thanks :-)